servicetalk-opentracing-zipkin-publisher/src/main/java/io/servicetalk/opentracing/zipkin/publisher/reporter/HttpReporter.java

```diff
@@ -0,0 +1,252 @@
+    }
+
+    /**
+     * A builder to create a new {@link HttpReporter}.
+     */
+    public static final class Builder {
+        private Codec codec = Codec.JSON_V2;
+        private final SingleAddressHttpClientBuilder<?, ?> clientBuilder;
+        private boolean disableBatching;
+        private int batchSizeHint = 16;
+        private int maxConcurrentReports = 32;
+        private Duration maxBatchDuration = ofSeconds(30);
+
+        /**
+         * Create a new {@link Builder} using the passed {@link SingleAddressHttpClientBuilder}.
+         *
+         * @param clientBuilder the collector SocketAddress
+         */
+        public Builder(final SingleAddressHttpClientBuilder<?, ?> clientBuilder) {
+            this.clientBuilder = clientBuilder;
+        }
+
+        /**
+         * Sets the {@link Codec} to encode the Spans with.
+         *
+         * @param codec the codec to use for this span.
+         * @return {@code this}
+         */
+        public Builder codec(Codec codec) {
+            this.codec = requireNonNull(codec);
+            return this;
+        }
+
+        public Builder maxConcurrentReports(final int maxConcurrentReports) {
+            this.maxConcurrentReports = maxConcurrentReports;
+            return this;
+        }
+
+        /**
+         * Configure batching of spans before sending it to the zipkin collector.
+         *
+         * @param batchSizeHint Hint of how many spans should be batched together.
+         * @param maxBatchDuration {@link Duration} of time to wait for {@code batchSizeHint} spans in a batch.
+         * @return {@code this}.
+         */
+        public Builder batchSpans(final int batchSizeHint, final Duration maxBatchDuration) {
+            if (batchSizeHint <= 0) {
+                throw new IllegalArgumentException("batchSizeHint: " + batchSizeHint + " (expected > 0)");
+            }
+            disableBatching = false;
+            this.batchSizeHint = batchSizeHint;
+            this.maxBatchDuration = requireNonNull(maxBatchDuration);
+            return this;
+        }
+
+        /**
+         * Disable batching of spans before sending them to the zipkin collector.
+         *
+         * @return {@code this}.
+         */
+        public Builder disableSpanBatching() {
+            disableBatching = true;
+            return this;
+        }
+
+        /**
+         * Builds a new {@link HttpReporter} instance with this builder's options.
+         *
+         * @return a new {@link HttpReporter}
+         */
+        public HttpReporter build() {
+            return new HttpReporter(this);
+        }
+    }
+
+    @Override
+    public void report(final Span span) {
+        if (closeInitiated) {
+            throw new IllegalStateException("Span: " + span + " reported after reporter " + this + " is closed.");
+        }
+        buffer.onNext(span);
+    }
+
+    @Override
+    public void close() {
+        awaitTermination(closeable.closeAsync().toFuture());
```

Comment:
Consider changing `closeable` type to `CompositeCloseable`. Then you can use `closeable.close()` which does the same.